### PR TITLE
feat: websocket documentation

### PR DESCRIPTION
**Pull request by Sorikairox**
Please check a preview of the doc here: https://docs-git-feat-websocket-sorikairoxs-projects.vercel.app

### CODE REVIEW

File: src/websockets/exception-filters.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### Exception filters
+
+The only difference between the HTTP [exception filter](/overview/exception-filters.md) layer and the corresponding web sockets layer is that instead of throwing `HttpException`, you should use return a valid message with `topic` and `data`.
+
+```typescript
+return { topic: 'error', data: { customMessage: 'not cool my friend' }};
+```
+
+With the sample above, Nest will handle the thrown exception and emit the `exception` message with the following structure:
```
Comment (marco-souza) [inline src/websockets/exception-filters.md]:
```suggestion
With the sample above, Danet will handle the thrown exception and emit the `exception` message with the following structure:
```

Also, I think the strucutre is missing

File: src/websockets/middlewares.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+### Middlewares
+
+There is no difference between the HTTP [middlewares](/overview/middlewares.md) layer and the corresponding web sockets layer. But at least you know they exist !
+
+#### Example
+
+Web sockets exception filters behave equivalently to HTTP exception filters. The following example uses a manually instantiated method-scoped filter. Just as with HTTP based applications, you can also use gateway-scoped filters (i.e., prefix the gateway class with a `@UseFilters()` decorator).
```
Comment (marco-souza) [inline src/websockets/middlewares.md]: It looks like this is from `exception-filters`. Should we remove it?

File: src/websockets/middlewares.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	async action(ctx: ExecutionContext, next: NextFunction) {
+		ctx.websocket!.send(
+			JSON.stringify({
+				topic: 'status',
+				data: {
+          middleware: 'SimpleMiddleware'
+        },
+			}),
```
Comment (marco-souza) [inline src/websockets/middlewares.md]: minor identation fix:

```suggestion
			JSON.stringify({
				topic: 'status',
				data: {
          				middleware: 'SimpleMiddleware'
        			},
			}),
```

File: src/websockets/controllers.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
+# Controllers
+
+Most of the concepts discussed elsewhere in this documentation, such as dependency injection, decorators, exception filters, pipes, guards and interceptors, apply equally to controllers. Wherever possible, Danet abstracts implementation details so that the same components can run across HTTP-based platforms and WebSockets. This section covers the aspects of Danet that are specific to WebSockets.
+
+In Danet, a websocket contrller is simply a class annotated with `@WebSocketController()` decorator. Danet does not use any third party library to handle websocket, we use the [WebApi WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API)
```
Comment (marco-souza) [inline src/websockets/controllers.md]:
```suggestion
In Danet, a websocket controller is simply a class annotated with `@WebSocketController()` decorator. Danet does not use any third party library to handle websocket, we use the [WebApi WebSocket](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API)
```

File: src/websockets/controllers.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+In general, each WebSocketController is listening on the same port as the **HTTP server** and accessible on the given endpoints. For example, if you want to connect on `ws://youwebsite.com/ws` do the following: 
+
+```typescript
+@WebSocketController('ws')
+```
+
+The controller is now listening, but we have not yet subscribed to any incoming messages. Let's create a handler that will subscribe to the `events` messages and respond to the user with the exact same data.
+
+```ts events.controller.ts
+@SubscribeMessage('events')
```
Comment (marco-souza) [inline src/websockets/controllers.md]:
```suggestion
@OnWebSocketMessage('events')
```

File: src/websockets/controllers.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+handleEvent(@Body() data: string): string {
```
Comment (marco-souza) [inline src/websockets/controllers.md]: let's remove the `string` type, as it's different from the actual return of the example

File: src/websockets/controllers.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+```ts
+const websocket = new WebSocket(
+			`ws://localhost:3000/ws`,
+		);
+websocket.onopen = (e) => {
+			websocket.send(
+				JSON.stringify({ topic: 'trigger', data: { obiwan: 'kenobi' } }),
+			);
+		};
```
Comment (marco-souza) [inline src/websockets/controllers.md]:
```suggestion
const websocket = new WebSocket(`ws://localhost:3000/ws`);

websocket.onopen = (e) => {
    websocket.send(
        JSON.stringify({ topic: 'trigger', data: { obiwan: 'kenobi' } }),
    );
};
```

File: src/websockets/controllers.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+## Topic Matching
+
+Under the hood, we use Hono smart router on message topic. This mean that you can have very specific message handler, as if you were building an HTTP endpoints. Some examples from Hono's documentation: 
+
+- '/user/:name'
+- '/posts/:id/comment/:comment_id'
+- '/api/animal/:type?'
+- '/post/:date{[0-9]+}/:title{[a-z]+}'
```
Comment (marco-souza) [inline src/websockets/controllers.md]: minor suggestion to make it a code snippet

```suggestion
- `'/user/:name'`
- `'/posts/:id/comment/:comment_id'`
- `'/api/animal/:type?'`
- `'/post/:date{[0-9]+}/:title{[a-z]+}'`
```

File: src/websockets/controllers.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+```ts events.controller.ts
+@OnWebSocketMessage('events')
+handleEvent(@Body('name') name: string) {
+  return { topic: 'hello', data: name };
+}
+```
+
+## Execution context
+
+When implementing any websocket features (such as `ExceptionFilters`, `Middleware` or `Guards`) and you get access to the `ExecutionContext`, you have access to two additionnal properties, `WebSocket`, representing the current socket, a `WebSocketTopic`, with the actual message topic.
```
Comment (marco-souza) [inline src/websockets/controllers.md]:
```suggestion
When implementing any websocket features (such as `ExceptionFilters`, `Middleware` or `Guards`) and you get access to the `ExecutionContext`, you have access to two additionnal properties, `webSocket`, representing the current socket, a `webSocketTopic`, with the actual message topic.
```

File: src/websockets/middlewares.md (hunk 2 above)
Comment (Sorikairox) [inline src/websockets/middlewares.md]:
```suggestion
```

File: src/websockets/controllers.md (hunk 6 above)
Comment (Sorikairox) [inline src/websockets/controllers.md]:
```suggestion
handleEvent(@Body() data: unknown): WebSocketPayload {
```

**Pull request merged**